### DISCUSSION TITLE
MacMini for UNO (and .NET MAUI)

### DISCUSSION THREAD

**Discussion by ward-horsfall** [unoplatform/uno#16782] [Q&A]
Hi,

I am looking at getting a Mac MINI M2/M2 Pro for development for the
IOS simulator and publishing the App to the Apple Appstore.

Specifically doing coding and debugging from the Visual Studio 2022 on Windows.

I am trying to work out which Mac Mini to buy:

https://www.apple.com/au/shop/buy-mac/mac-mini

8-Core CPU - 10-Core GPU - 8GB Unified Memory - 256GB SSD Storage¹
A$999.00

8-Core CPU - 10-Core GPU - 8GB Unified Memory - 512GB SSD Storage¹
A$1299.00

10-Core CPU - 16-Core GPU - 16GB Unified Memory - 512GB SSD Storage¹
A$1999.00

Problem is I don't want to buy something that that will under perform.

Specifically, run too slow to be frustrating and/or run out of storage.

I am looking for some general advice,

Thanks

Ward

**Comment by agneszitte** [comment]
cc @jeromelaban, @spouliot

**Comment by spouliot** [comment]
Disclaimer: My main computers are Macs (a M1 mini and an M2 air) and I something use a Windows (arm64) computer. I don't do remote builds (like you seem to plan) because I know how frustrating they can be.

My general advice for Macs is that, because you can't update them (at least the CPU and RAM), go for the biggest your budget allow. It will last longer and you'll appreciate the extra performance - not that any M2 will underperform to builds iOS apps (considering it will be dedicated to that task).

Now be aware that VS 2022 remote builds can be network intensive, a lot of data will move back and forth your Windows and Mac computers (since different parts of the build is done on each side). Avoid using WiFi between your two computers. A gigabit switch should be fine (that's what I use) and  there's an option for 10G on the mini (in case you already a 10G network).

Finally if you fall in love with your Mac (it's been known to happen) and still want/need VS 2022 then you can virtualize Windows (e.g. with VMWare Fusion or Parallel) and run VS inside it (as it run as an arm64 binary). However you might want 24GB or 32GB (for M2Pro) of RAM for such setup.
